tests/foreman/cli/test_computeresource_vmware.py

```diff
@@ -150,6 +152,33 @@ def test_positive_provision_end_to_end(
         != 'Pending installation',
         timeout=1800,
         delay=30,
+        silent_failure=True,
+        handle_exception=True,
     )
+    vcenter8_issue = provision_method == 'bootdisk' and vmware == '8' and is_open('BZ:2255264')
     host_info = sat.cli.Host.info({'id': host['id']})
-    assert host_info['status']['build-status'] == 'Installed'
+    assert (
+        host_info['status']['build-status'] == 'Pending installation'
+        if vcenter8_issue
+        else 'Installed'
+    )
+
+    # Perform OS version check and check if root password is properly updated
+    provisioning_host = ContentHost(host_info['network-interfaces'][0]['ipv4-address'])
+    expected_rhel_version = host_info['operating-system']['operating-system'][-3:]
+    if int(expected_rhel_version.split('.')[0]) >= 9:
+        assert (
+            provisioning_host.execute(
+                'echo -e "\nPermitRootLogin yes" >> /etc/ssh/sshd_config; systemctl restart sshd'
+            ).status
+            == 0
+        )
+    host_ssh_os = sat.execute(
```

Comment:
Why not directly `provisioning_host.execute()`?

Reply:
I remember this, we are doing this to check whether the password is set correctly on the provisioned host

Reply:
Understood. Put it in a comment, please!

Reply:
I added comment that exists at top already :)
`# Perform OS version check and check if root password is properly updated`